add section why cache limitation is necessary with examples #70

ahus1 opened the pull request:
This tries to document the necessary web server setup regarding caching. This should close #70

jeffposnick commented on README.md:
```diff
@@ -22,6 +22,12 @@ $ npm run dev
+## What's not Included
+
+* You'll need to configure your web server to add HTTP headers to prevent caching of critical service worker files.
+If you don't do this, you might end up with browsers that never update the cached content.
```
It's not (quite) as bad as that.

Can you update the "never" to say that there might be a delay of up to 24 hours? And link to https://stackoverflow.com/questions/38843970/service-worker-javascript-update-frequency-every-24-hours/38854905#38854905

jeffposnick commented on README.md:
```diff
@@ -63,3 +69,46 @@ You can fork this repo to create your own boilerplate, and use it with `vue-cli`
+When you run the production build and deploy the application to a web server, the browser will call the service worker on the first visit and will download and cache all resources. 
+When the browser accesses the site a second time, it will use the cached resources.
+To notice changes in your app it is crucial that the browser downloads the latest version `service-worker.js` from your server.
+The browser will only do this when the HTTP Header of the service worker indicates that a previous download has expired.
```
Can you rephrase "HTTP Header" to "[Cache-Control](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/http-caching) header associated with `service-worker.js`"

jeffposnick commented on README.md:
```diff
@@ -63,3 +69,46 @@ You can fork this repo to create your own boilerplate, and use it with `vue-cli`
+The browser will only do this when the HTTP Header of the service worker indicates that a previous download has expired.
+If a new service worker is available, it will download the new contents in the background.
+Once the download is complete, the default behavior is that the new content will be shown on the next visit of the site. 
+You can change the template's `service-worker-prod.js` to display the message to the user to reload the page, or issue a `window.location.reload()` automatically to trigger an update depending on your application's needs.
```
`service-worker-prod.js` => [`service-worker-prod.js`](https://github.com/vuejs-templates/pwa/blob/master/template/build/service-worker-prod.js)

jeffposnick commented on README.md:
```diff
@@ -63,3 +69,46 @@ You can fork this repo to create your own boilerplate, and use it with `vue-cli`
+You can change the template's `service-worker-prod.js` to display the message to the user to reload the page, or issue a `window.location.reload()` automatically to trigger an update depending on your application's needs.
+
+Depending on your setup, you might want to have a short caching time for your service worker or no caching at all.
+At minimum, you should add caching headers for `service-worker.js` and `index.html`.
```
Can you rephrase "you should add caching headers" to make it clear that the headers you add should disable caching?

jeffposnick commented on README.md:
```diff
@@ -63,3 +69,46 @@ You can fork this repo to create your own boilerplate, and use it with `vue-cli`
+
+Depending on your setup, you might want to have a short caching time for your service worker or no caching at all.
+At minimum, you should add caching headers for `service-worker.js` and `index.html`.
+You should also add caching headers for all files that are in the `static` folder of this template.
```
Link to https://jakearchibald.com/2016/caching-best-practices/

ahus1 commented:
@jeffposnick - Hello Jeff, thank you for the review. I've added the new content. I've also move the some of the content I added to README.md to a file docs/prevent_caching.md. 

Please review again.

jeffposnick commented:
Thanks!
